components/module-components/Contact/ContactForm.tsx

```diff
@@ -0,0 +1,125 @@
+'use client';
+
+import React from 'react';
+import { ToastContainer, toast } from 'react-toastify';
+import 'react-toastify/dist/ReactToastify.css';
+import { useFormik } from 'formik';
+import * as Yup from 'yup';
+import style from './contact.module.scss';
+import { ContactFormData, submitContactForm } from './SubmitContactForm';
+
+interface ContactFormProps {
+  inputClassName?: string;
+  textareaClassName?: string;
+  buttonClassName?: string;
+}
+
+const ContactForm: React.FC<ContactFormProps> = ({
+  inputClassName,
+  buttonClassName,
+  textareaClassName,
+}) => {
+  const contactEmail = /^[^\s@]+@[^\s@]+\.[^\s@]+$/;
+
+  const formik = useFormik({
+    initialValues: { username: '', email: '', message: '' },
+    validationSchema: Yup.object({
+      username: Yup.string()
+        .matches(/^[\u0400-\u04FFа-џA-Za-z-]+$/, 'Невалидно име')
+        .required('Задолжително внесете име'),
+      email: Yup.string()
+        .email('Невалидна емаил адреса!')
+        .required('Задолжително внесете емаил адреса')
+        .matches(contactEmail, 'Погрешен емаил формат'),
+      message: Yup.string().required('Пораката е задолжителна'),
+    }),
+    onSubmit: async (values, { resetForm }) => {
+      try {
+        const formData: ContactFormData = {
+          first_name: values.username,
+          last_name: '',
+          email: values.email,
+          subject: '',
+          message: values.message,
+          'cf-turnstile-response': '',
```

Comment:
you will need the changes @Mikivitkovski made, because you need to send here the Cloudflare Turnstile ReCaptcha token. (that PR is still not merged)